docs/dialect.md

```diff
@@ -60,7 +61,58 @@ return-type | Y | Indicates the return type of the function. For a list of allow
 
 - __date-function element__  
 The <span style="font-family: courier new">date-function</span> element is a specialized variant of <span style="font-family: courier new">function</span>. In addition to the base formula, you can specify one or more datepart formulas, which are used instead of the generic formula when available.  
-The function <span style="font-family: courier new">name</span> must be one of these: DATEADD, DATEDIFF, DATEFORMAT, DATENAME, DATEPARSE, DATEPART, DATETRUNC.  
+The function <span style="font-family: courier new">name</span> must be one of these: DATEADD, DATEDIFF, DATEFORMAT, DATENAME, DATEPARSE, DATEPART, DATETRUNC.   
+
+  -  __DATEPART without custom start of week__
+    ```xml
+        ...
+        <date-function name='DATEPART' return-type='int'>
+          <formula>CAST(TRUNC(EXTRACT(%1 FROM %2)) AS INTEGER)</formula>
+          <formula part='weekday'>(1 + CAST(EXTRACT(DOW FROM %2) AS INTEGER))</formula>
+          <formula part='week'>CAST(FLOOR((7 + EXTRACT(DOY FROM %2) - 1 + EXTRACT(DOW FROM DATE_TRUNC(&apos;YEAR&apos;, %2))) / 7) AS INTEGER)</formula>
+          <argument type='localstr' />
+          <argument type='datetime' />
+        </date-function>
+        ...
+    ```
+
+    Here the first argument (%1) is grabbed from the `<date-part-group>` in the dialect file. It is used with generic (non-part-specific) date function formulas.
+    A date-part-group can apply to one or more date functions, denoted by date-function child elements. If none are specified, the group acts as the default.
+    The name attribute specifies a Tableau date part, while the value attribute contains the date part string literal to use in corresponding date functions.
+
+    ```xml
+        ...
+          <date-part-group>
+            <date-function name='DATEPART' />
+            <part name='year' value='YEAR' />
+            <part name='quarter' value='QUARTER' />
+            <part name='month' value='MONTH' />
+            <part name='week' value='WEEK' />
+            <part name='weekday' value='DOW' />
+            <part name='dayofyear' value='DOY' />
+            <part name='day' value='DAY' />
+            <part name='hour' value='HOUR' />
+            <part name='minute' value='MINUTE' />
+            <part name='second' value='SECOND' />
+          </date-part-group>
+        ...
+    ```
+    A single date function can have multiple overloaded functions with different parameters. For example here, the DatePart function is overloaded with argument for the start of the week.
```

Comment:
I think let's just describe this as something like: To support Tableau's Custom Start of Week functionality each of the following: DATEDIFF, DATENAME, DATEPART, DATETRUNC need to also have an overloaded form with an additional `<argument type='localstr' />`.